### PR TITLE
TelemetryLoggerMiddleware 

### PR DESCRIPTION
**Pull request by daveta**
Middlware component that uses IBotTelemetryClient to log events when messages are received/sent/updated or deleted in the bot.

Additional Details
https://github.com/daveta/analytics/blob/master/telemetry_enhancements/TelemetryEnhancements.md

js: https://github.com/Microsoft/botbuilder-js/pull/812

### CODE REVIEW

**Comment by ryanlengel** [comment]
Should all telemetry properties follow the same casing? This middleware & the luis/qna samples use lower camel case. The waterfall telemetry & initializer is upper camel case.

File: libraries/Microsoft.Bot.Builder/TelemetryConstants.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder
+{
+    public static class TelemetryConstants
+    {
+        public const string ChannelIdProperty = "channelId";
```
Comment (ryanlengel) [inline libraries/Microsoft.Bot.Builder/TelemetryConstants.cs]: Should all telemetry properties follow the same casing? This middleware & the luis/qna samples use lower camel case. The waterfall telemetry & initializer is upper camel case.

Comment (daveta) [reply libraries/Microsoft.Bot.Builder/TelemetryConstants.cs]: Argh.  
The Initializer properties begin with lowercase, so we're already inconsistent.  I'm leaning towards aligning with Initializer and begin with lowercase character - any thoughts?

Tempted to lowercase waterfall telemetry, but that's actually a breaking change.

---
In reply to: [263166401](https://github.com/Microsoft/botbuilder-dotnet/pull/1420#discussion_r263166401) [](ancestors = 263166401)

Comment (ryanlengel) [reply libraries/Microsoft.Bot.Builder/TelemetryConstants.cs]: I agree with lowercase

**Pull request merged**